Peer Review: Maia Chat (#9100)
mendix/docs: content/en/docs/refguide/modeling/mendix-ai-assistance/maia-chat.md

--- 
title: "Maia Chat" 

> linktitle: "Chat" 

Author: I think "Maia Chat" is short enough for the linktitle. We don't need to shorten it further. And I think "Maia" is important to keep as it is the name of the tab, the name we give to the ✨ icon, and something we use in marketing.

Reviewer: Not including "Maia" in the linktitle is actually a conscious decision. We did this because it's already under the umbrella term of **Mendix AI Assistance (Maia)** in the left navigation menu. This is basically the only place we use "Chat" instead of "Maia Chat" and is actually still in alignment with what's in the UI:
![image](https://github.com/user-attachments/assets/18bc9ed2-c3ed-4d74-801a-908587cc3aec)

For more information, see the style guide that I wrote for **Mendix AI Assistance (Maia)**: https://mendix.atlassian.net/wiki/spaces/RNDHB/pages/2109145362/Other+Mendix+Terms#Mendix-AI-Assistance-(Maia).

Author: Great - thanks for the explanation.

url: /refguide/maia-chat/ 
weight: 5 
description: "Describes the features in Maia Chat." 
aliases: 

>     - /refguide/mendix-chat/ 

Author: The alias is the same as the URL, so it isn't needed
```suggestion
```

Reviewer: The URL is `/refguide/maia-chat/ `and the alias is `/refguide/mendix-chat/`. They are different. I'll ignore this commit suggestion :)

Author: Whoops - not reading carefully. Yes, we definitely need this.

> #If moving or renaming this doc file, implement a temporary redirect and let the respective team know they should update the URL in the product. See Mapping to Products for more details. 

Author: Maia Chat hasn't been added to the mapping document so we don't know where the link is coming from or who to ask to update the URL.

Reviewer: Good one! I will add this to the mapping document.

--- 
 
## Introduction  
 
{{% alert color="info" %}} 

> MendixChat was rebranded to Maia Chat and released in GA in Studio Pro 10.12.0. It is also available in the Mendix Portal. 

Author: I wonder if this would be better the other way around. It is a bit strange that the first word in the document is "MendixChat" when customers are expecting to read about "Maia Chat". Perhaps
```suggestion
Maia Chat was released as GA in Studio Pro 10.12.0. Between Mendix 10.6 and 10.11 it was available as a beta and called MendixChat. Maia Chat is also available in the Mendix Portal. 
```

Reviewer: Good suggestion! I revised the text a bit based on your suggestion: https://github.com/mendix/docs/pull/9100/commits/53872a1564664d799e8cf69e6988741f91c7aeed.

{{% /alert %}} 
 
{{% alert color="info" %}} 
To use Maia Chat, an internet connection and signing in to Studio Pro are required. 
{{% /alert %}} 
 
{{% alert color="info" %}} 
Maia Chat does not use any project, customer, or company information. It only uses the data that you have entered in the Chat. Please do not enter any personal or sensitive data! 
{{% /alert %}} 
 

> Maia Chat is a built-in chat interface in the Mendix Platform powered by Generative AI. Developers can ask questions about all aspects of Mendix, including how to apply concepts, best practices, governance tools, and development patterns. It is based on large language models (LLMs) using data from various resources, such as [Mendix Docs](/), [Mendix Community](https://community.mendix.com/), and [Mendix Academy](https://academy.mendix.com/). 

Author: I think we can use a lowercase "g" for generative here?
```suggestion
Maia Chat is a built-in chat interface in the Mendix Platform powered by generative AI. Developers can ask questions about all aspects of Mendix, including how to apply concepts, best practices, governance tools, and development patterns. It is based on large language models (LLMs) using data from various resources, such as [Mendix Docs](/), [Mendix Community](https://community.mendix.com/), and [Mendix Academy](https://academy.mendix.com/). 
```

 
## Using Maia Chat 
 
There are two ways to access Maia Chat:  
 
* Click the **Maia** ({{% icon name="sparkles" %}}) icon to the right of the top bar. 
* Go to the **View** menu and click **Maia**. 
 
The chat interface appears at the right side of Studio Pro under the **Maia** pane: 
 

> {{< figure src="/attachments/refguide/modeling/mendix-ai-assistance/maia-chat/maia-chat-interface.png" width="300px">}} 

Author: The Maia pane now has **Chat** and **Learn**. I think we need an updated image and, perhaps, something about **Learn** (probably with a link to more information rather than a full description here.)
![image](https://github.com/user-attachments/assets/7ea049bb-2ee9-407c-889f-04b81b6c3f44)

Reviewer: Good catch. Updated the screenshot and also added some information about the **Learn** tab in this commit: https://github.com/mendix/docs/pull/9100/commits/fcfb8878a478eddbd0f0b18e62da4a9ae900af0d/

 
### An Example of Interacting with Maia Chat 
 

> You can start using Maia Chat by clicking one of the demo questions. You can also start by asking your own question in the chatbox.  

Author: Perhaps better not to use "start" twice. Perhaps just
```suggestion
You can start using Maia Chat by clicking one of the demo questions. Alternatively, you can ask your own question in the chatbox.  
```
or something else.

Not sure I like "chatbox" as a term. Do we actually have to call it anything, or is it obvious that this is the place to type. Perhaps "input area"? Or is this what the Maia team want to call it?

Reviewer: I've removed "in the chatbox".

 
The following steps present an example of how you can interact with Maia Chat by asking your own question: 
 

> 1. Type the following question in the chatbox: `When should I use a calculated attribute?`. 

Author: See note above - do we need to say "chatbox"?

Reviewer: Removed it :)

> 2. Press <kbd>Enter</kbd> or the **Ask** button to send the question. You will then see the following: 

Author: We normally use "click" for buttons, and I'm not sure we have to call it a button?
```suggestion
2. Press <kbd>Enter</kbd> or click **Ask** to send the question. You will then see the following: 
```

Reviewer: Indeed, I don't think we have to mention "button" here.

    * An answer generated by Maia Chat 

>     * Top relevant references from various resources like Mendix documentation, Mendix Community, and Academy learning paths 

Author: You will also see the feedback buttons 😀

Reviewer: They were in a separate section called **Studio Pro Features**. 
However, since this doc is under the Studio Pro refguide, I do think those features should be mentioned together.
I took this opportunity to restructure the whole doc. I removed the example section and added a **Maia Chat Features in Studio Pro** section where all the features are put under this section. This also makes it easier to maintain the doc if any new features are added in the future :) Please let me know if you have any feedback.
Here is the commit where I made all these changes: https://github.com/mendix/docs/pull/9100/commits/44d330306eeea9138c94f00276e45b29c46c8639.

>     * Top relevant references from various resources like Mendix documentation, Mendix Community, and Academy learning paths 

Author: Perhaps "links to" rather than "references from"? Feels more active that way around?
```suggestion
    * Links to top relevant resources like Mendix documentation, Mendix Community, and Academy learning paths 
```

 
3. Click the thumbs up icon ({{% icon name="thumbs-up" %}}) or the thumbs down icon ({{% icon name="thumbs-down" %}}) to give your feedback on an answer.  
 

> 4. Click the add feedback icon ({{% icon name="notes-paper-edit" %}}) to provide a more detailed feedback. A detailed **Feedback** section shows below the answer: 

Author: Don't need an article here:
```suggestion
4. Click the add feedback icon ({{% icon name="notes-paper-edit" %}}) to provide more detailed feedback. A detailed **Feedback** section shows below the answer: 
```

 
    * You can rate the **correctness**, **completeness**, and **readability** of the provided answer. 
 
    * You can also provide **Additional feedback** if you want to.  
 
      {{< figure src="/attachments/refguide/modeling/mendix-ai-assistance/maia-chat/feedback-section.png" width="250px" >}}  
 

> 5. Click the **Submit feedback** button to submit your feedback. Afterwards, you will be brought back to the main chat interface where you can continue asking questions. 

Author: These things are optional - perhaps we should mention that, but encourage them to use if by explaining that the responses are used to improve Maia's responses?

Reviewer: They are now in a separate sub-section called **Giving Feedback** in the **Maia Chat Features in Studio Pro** section.

> 6. Click **New chat** at beginning of your chat if you want to clear the messages and start a new conversation without reference to your current chat (this option was added in Studio Pro 10.7.0). 

Author: Perhaps "which will not reference your current chat" would be clearer?
```suggestion
6. Click **New chat** at beginning of your chat if you want to clear the messages and start a new conversation which will not reference your current chat (this option was added in Studio Pro 10.7.0). 
```
Are we sure this was 10.7? The Follow-Up questions section says that this was introduced in 10.8, so the New Chat button seems redundant in 10.7?

Also, since we haven't talked about context yet, perhaps link to **Asking Follow-Up Questions** here, or add an example of a follow-up question?

Reviewer: Yes, it was introduced in 10.7 🙂

Regarding: 

> Also, since we haven't talked about context yet, perhaps link to Asking Follow-Up Questions here, or add an example of a follow-up question?

Please see my reply here: https://github.com/mendix/docs/pull/9100#discussion_r1973883196.

 
#### Studio Pro Features 
 
Maia Chat in Studio Pro has additional features which are indicated by the following: 
 
* A copy icon ({{% icon name="copy" %}}) which allows you to copy the given answer to a clipboard (this option was added in 10.7.0) 
* Feedback icons 
* New chat option 
   
    {{< figure src="/attachments/refguide/modeling/mendix-ai-assistance/maia-chat/maia-chat-example.png" width="350px" >}} 
     

>     {{% alert color="info" %}}In Studio Pro 10.7.0 and above, you can also see the Mendix version of the references, the level required if the reference comes from Mendix Academy, and the date and answer status if the reference comes from the Mendix Community.{{% /alert %}} 

Author: Not sure if it is the level _required_. Perhaps just
```suggestion
    {{% alert color="info" %}}In Studio Pro 10.7.0 and above, you can also see the Mendix version of the references, the level of the learning path if the reference comes from Mendix Academy, and the date and answer status if the reference comes from the Mendix Community.{{% /alert %}} 
```

 
### Asking Follow-Up Questions 
 
In Studio Pro 10.8.0 and above, Maia Chat supports asking follow-up questions based on your previous question and the answer you have received. For example, you can ask it to summarize the latest answer you received by typing the following in the chat: *Can you summarize that?*. 
 
{{% alert color="info" %}} 
 
Maia Chat will lose track of the conversation history if you exit Studio Pro. 
 

> {{% /alert %}} 

Author: I wonder if we should mention this feature during our example, above. I think it is very powerful. Could add it between steps 5 and 6? I don't mean remove this section, just add its use as part of the example?

Reviewer: Please see my reply here: https://github.com/mendix/docs/pull/9100#discussion_r1973883196.

You can start using Maia Chat by clicking one of the demo questions. Alternatively, you can ask your own question, and press <kbd>Enter</kbd> or click **Ask** to send the question. For information on the supported shortcut keys in Maia Chat, see the [Maia Chat Shortcut Keys](/refguide/keyboard-shortcuts/#maia-chat-shortcuts) section in *Keyboard Shortcuts*.

For more information on Maia Chat features in Studio Pro, see the sub-sections below.

### Maia Chat Features in Studio Pro

After a question is sent to Maia Chat, you will see the following in the chat interface:

* An answer generated by Maia Chat
* Links to top relevant resources like Mendix documentation, Mendix Community, and Academy learning paths

    {{% alert color="info" %}}In Studio Pro 10.7.0 and above, you can also see the Mendix version of the references, the level of the learning path if the reference comes from Mendix Academy, and the date and answer status if the reference comes from the Mendix Community.{{% /alert %}}

* A copy icon ({{% icon name="copy" %}}) which allows you to copy the given answer to a clipboard (available in Studio Pro 10.7.0 and above)
* Feedback icons

#### Giving Feedback

> Mendix encourages giving feedback on the answers generated by Maia Chat. Feedback will be used to further improve Maia Chat.

Author: This is really nice.
